libcudacxx/include/cuda/std/detail/libcxx/include/__type_traits/promote.h

```diff
@@ -19,6 +19,10 @@
 #include "../__utility/declval.h"
 #include "../cstddef"
 
+#ifdef __cuda_std__
```

Comment:
comment: Check with @gevtushenko if we need to guard this include based on other things like CTK version. I seem to recall CUB's include of the `cuda_fp16.h` header has a more complex check.

Reply:
suggestion: If so, this would be a good opportunity to add a unified `__CCCL_HAS_CUDAFP16`.

Reply:
Yeah, this will need a more sophisticated check.

Reply:
Actually, no, it seems that `cuda_bf16.h` appeared in CUDA 11. What was the CUB issue?

Reply:
https://github.com/NVIDIA/cub/issues/478. The problem with bf16 headers is that they are not redistributable based on CUDA EULA, so in many runtime environments (including NV's own runtime dockers) this functionality would be explicitly opted out.

Reply:
Gotcha. I'll figure something out.

Reply:
@jrhemstad turns out we have different requirements here from CUB, but in the current logic of disabling the specializations for half and bfloat I also check for the CUB opt out macro (and, indeed, had to, because the CUB test for not including it when the macro is defined also happens to test this aspect of libcu++).